test/test_sparse.py

```diff
@@ -2323,6 +2326,10 @@ def test_factory_type_inference(self):
         self.assertEqual(torch.float32, t.dtype)
         t = torch.sparse_coo_tensor(torch.tensor(([0], [2])), torch.tensor([1.], dtype=torch.float64))
         self.assertEqual(torch.float64, t.dtype)
+        t = torch.sparse_coo_tensor(torch.tensor(([0], [2])), torch.tensor([1.], dtype=torch.cfloat))
```

Comment:
consider having non zero imag value in the tensor e.g., `torch.tensor([2.3-4.5j], dtype=torch.cfloat)`